src/renderer/pages/Settings/Networks/Networks.tsx

```diff
@@ -41,15 +44,21 @@ export const Networks = () => {
     setTimeout(() => navigate(Paths.SETTINGS), DEFAULT_TRANSITION);
   };
 
-  const { inactive, active } = Object.values(connections).reduce<Record<'inactive' | 'active', ExtendedChain[]>>(
-    (acc, c) => {
-      if (!includes(c.name, query)) return acc;
-
-      if (c.connection.connectionType === ConnectionType.DISABLED) {
-        acc.inactive.push(c);
-      }
-      if (c.connection.connectionStatus !== ConnectionStatus.NONE) {
-        acc.active.push(c);
+  const { inactive, active } = Object.values(chains).reduce<Record<'inactive' | 'active', ExtendedChain[]>>(
+    (acc, chain) => {
+      if (!includes(chain.name, query)) return acc;
+
+      const connection = connections[chain.chainId];
+      const extendedChain = {
+        ...chain,
+        connection,
+        connectionStatus: connectionStatuses[chain.chainId],
+      };
+
+      if (connection?.connectionType === ConnectionType.DISABLED) {
+        acc.inactive.push(extendedChain);
+      } else {
+        acc.active.push(extendedChain);
```

Comment:
We could also use `partition`

Reply:
@valentunn could you tell more about `partition` usage?

Reply:
@tuul-wq it is used to separate a collection into two subcollections based on boolean condition